yun/libraries/Parse/src/internal/ParseClient.cpp

```diff
@@ -89,82 +93,89 @@ ParseResponse ParseClient::sendRequest(const char* httpVerb, const char* httpPat
-
-	requestClient.addParameter("-v");
-	requestClient.addParameter(httpVerb);
-	requestClient.addParameter("-e");
-	requestClient.addParameter(httpPath);
-	if (requestBody != "") {
-		requestClient.addParameter("-d");
-		requestClient.addParameter(requestBody);
-	}
-	if (urlParams != "") {
-		requestClient.addParameter("-p");
-		requestClient.addParameter(urlParams);
+  requestClient.begin("parse_request");  // start a process that launch the "parse_request" command
+
+  requestClient.addParameter("-v");
+  requestClient.addParameter(httpVerb);
+  requestClient.addParameter("-e");
+  requestClient.addParameter(httpPath);
+  if (requestBody != "") {
+    requestClient.addParameter("-d");
+    requestClient.addParameter(requestBody);
+  }
+  if (urlParams != "") {
+    requestClient.addParameter("-p");
+    requestClient.addParameter(urlParams);
     requestClient.runAsynchronously();
-	} else {
-	  requestClient.run(); // Run the process and wait for its termination
-	}
+  } else {
+    requestClient.run(); // Run the process and wait for its termination
+  }
 
-	ParseResponse response(&requestClient);
-	return response;
+  ParseResponse response(&requestClient);
+  return response;
 }
 
 bool ParseClient::startPushService() {
-	pushClient.begin("parse_push");  // start a process that launch the "parse_request" command
-	pushClient.runAsynchronously();
-
-	while(1) {
-		if(pushClient.available()) {
-			// read the push starting result
-			char c = pushClient.read();
-			while(pushClient.available()) {
-				pushClient.read();
-			}
-			if (c == 's') {
-                 pushClient.write('n'); // send a signal that ready to consume next available push
-				return true;
-			} else {
-				return false;
-			}
-		}
-	}
+  pushClient.begin("parse_push");  // start a process that launch the "parse_request" command
+  pushClient.runAsynchronously();
+
+  while(1) {
+    if(pushClient.available()) {
+      // read the push starting result
+      char c = pushClient.read();
+      while(pushClient.available()) {
+        pushClient.read();
+      }
+      if (c == 's') {
+        pushClient.write('n'); // send a signal that ready to consume next available push
+        return true;
+      } else {
+        return false;
+      }
+    }
+  }
 }
 
 ParsePush ParseClient::nextPush() {
-	ParsePush push(&pushClient);
-	return push;
+  ParsePush push(&pushClient);
+  return push;
 }
 
 bool ParseClient::pushAvailable() {
-    pushClient.write('n'); // send a signal that ready to consume next available push
-    if(pushClient.available()) {
-        return 1;
-    } else {
-        return 0;
-    }
+  pushClient.write('n'); // send a signal that ready to consume next available push
+  if(pushClient.available()) {
+    return 1;
+  } else {
+    return 0;
+  }
 }
 
 void ParseClient::stopPushService() {
-	pushClient.close();
+  pushClient.close();
 }
 
 void ParseClient::end() {
-	stopPushService();
+  if(installationId) {
+    delete[] installationId;
+  }
```

Comment:
Should we also set installationId/sessionToken to NULL after freeing the memory?